### DISCUSSION TITLE
How to install both arm64-osx and x64-osx versions of packages?

### DISCUSSION THREAD

**Discussion by MarkCallow** [microsoft/vcpkg#20186] [Q&A] [answered]
In my CI builds which are done on Intel-based Macs I need to build Intel and Apple Silicon versions of my apps. (My first choice was to make universal builds but those are not yet supported here.) To do this I need to install in the CI environment both x64 and arm64 versions of the library packages on which my apps depend. Is this possible?

Related question. In the list at https://vcpkg.io/en/packages.html, the packages I need do not state arm64-osx support but neither do any others that I see. Is arm64-osx supported by any package. I  didn't see any stating such support. I tried to search for "arm64-osx" but the search function at that url appears to work only on package names. Perhaps it is just that the list has not been updated. Is it possible to specify arm64-osx for these builds?

Aside: "osx" is no longer an accurate name as Apple has moved on beyond version 10. The name should be changed to "{x,arm)64-macos."

**Comment by autoantwort** [comment, marked as answer]
1. If you are using classic mode, you can simply do `vcpkg install zlib:x64-osx zlib:arm64-osx`. If you are using manifest mode you could do something like `vcpkg install --triplet=x64-osx --x-install-root=x64` and `vcpkg install --triplet=arm64-osx --x-install-root=arm64`. 

2. arm64-osx is not tested in CI, so you have to test yourself if a package builds on arm64 or not

3. This is known but kept for backwards compatibility, otherwise you break the would if you simply rename it.

**Reply by MarkCallow** [reply]
Thank you @autoantwork. The command given in 1. works. The `--x-install-root` is not necessary though. Without it stuff is installed in, e.g. `installed/x64-osx` in the same directory as the `vcpkg` binary. With it stuff is installed in``installed/x64/x64-osx`.
